include/utils.h

```diff
@@ -0,0 +1,11 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#include "blake2s.h"
+
+uint32_t rotr32(const uint32_t w, const unsigned c);
```

Comment:
These functions should be macros because they're used a lot. You could as well make them inline, but that makes problems.